DAQ_System/Source/main.cpp

```diff
@@ -1,23 +1,140 @@
 #include "mbed.h"
-#include <iostream>
+#include <cstdio>
+#include <cstdlib>
+
 
 #include "L3GD20H.h"
 // Found in: https://os.mbed.com/users/julioefajardo/code/L3GD20H/docs/tip/classL3GD20H.html
 
 
+#define sensitivity (.00875 )// this is using a sensitivity of 250 dps which is equivalent to .00875
+#define DPS_TO_RadS (3.14159/180) // this is the conversion factor for degrees per second to radians per second all you have to do is multiply the callibrated data or the raw data by the conversion factor
+L3GD20H gyro(I2C_SDA, I2C_SCL);
+
+void averageOffsetCalc(short averageOffset[], short sum[]);
+void degreesPerSecondConversion(short degreesPerSecond[],short offset[], short gyroData[]);
+void radiansPerSecondConversion(float radiansPerSecond[],short degreesPerSecond[]);
+
+
 // main() runs in its own thread in the OS
 int main()
 {
-    L3GD20H Gyro(I2C_SDA, I2C_SCL);
 
+    
     short gyroData[3] = {0, 0, 0};
 
-    while (true) {
-        Gyro.read(gyroData);
-        std::cout << "It's Gyroin Time"
-                  << "\tX: " << gyroData[0]
-                  << "\tY: " << gyroData[1]
-                  << "\tZ: " << gyroData[2] << std::endl;
-        ThisThread::sleep_for(500ms);
+    // for the 3 arrays below the first element will deal with purely the x-axis the second element is y and the last element is z. These will be used to calculate an average
+    // that will be used for the offeset for the callibration of the gyroscope. Int n is used to control the sample size for the average.
+    short sum[3] = {0}; 
+    short offset[3] = {0}; 
+    short degreesPerSecond[3] = {0};
+    float radiansPerSecond[3] = {0};
+
+
+    // logical conditions to determine the branches that will exeute in the code
+    bool gyroStatus;
+    bool offsetStatus = false;
+    
+ 
+    while (true) 
+    {
```

Comment:
The **opening** curly brace should be in-line with the statement (do not dedicate a whole line to it). Do this for the rest of the code as well